### PR TITLE
Use arrow functions in library_pthread.js. NFC

### PR DESCRIPTION
**Pull request by sbc100**

### CODE REVIEW

**Comment by RReverser** [comment]
Nice.

FWIW if you haven't tried yet you can use something like [ast-grep](https://ast-grep.github.io/guide/rule-config.html) to find-replace all functions with return body with arrow functions in the entire library JS codebase.

It uses tree-sitter which does loose parsing to correctly work in editors (as you edit code, it's usually not valid but still has to be parsed) so it even works reasonably well in Emscripten's custom flavour of JS with all the preprocessing directives.

The find-replace matches will still need manual review to ensure there are no usages of `this` and that it does work correctly with preprocessing conditions, but still should be much faster than upgrading libs manually.

**Comment by RReverser** [comment]
E.g. here's what ast-grep can do for this particular file / pattern:

image

File: src/library_pthread.js [hunk 1]
```diff
@@ -169,9 +169,7 @@ var LibraryPThread = {
 #endif
 
 #if !MINIMAL_RUNTIME
-    setExitStatus: (status) => {
-      EXITSTATUS = status;
-    },
+    setExitStatus: (status) => EXITSTATUS = status,
```
Comment (RReverser) [inline src/library_pthread.js:172]: It probably doesn't matter for how these particular functions are used, but note that such change changes the return type / value as now `setExitStatus` will return the exit status and `registerTLSInit` will return the new length of array.

Just wanted to confirm it's intended.

Comment (sbc100) [reply src/library_pthread.js:172]: I do know about that yes, and my assumption is that going from returning nothing to returning something in these cases is likely harmless.  

Perhaps there will be cases where it does matter but I don't see it mattering in these cases.

**Comment by RReverser** [comment]
Another note: please add this to git-ignore-revs file.

**Comment by sbc100** [comment]
> E.g. here's what ast-grep can do automatically for this particular file / pattern:
> 
> image
> 
> And many more matches with auto-replacements across the rest of the codebase:
> 
> image

Wait you are developing on windows now? :)

**Comment by sbc100** [comment]
> E.g. here's what ast-grep can do automatically for this particular file / pattern:
> 
> image
> 
> And many more matches with auto-replacements across the rest of the codebase:
> 
> image

Thats awesome!  do you want to send a followup applying all of those changes?

**Comment by RReverser** [comment]
> Wait you are developing on windows now? :)

Heh I almost always was; I'm just doing a lot of work inside WSL for Linux-specific stuff. I used to do that for Emscripten too, but recently trying to see if I can work on Emscripten from Windows again. Still some nasty issues with our scripts, but otherwise not too bad.

**Comment by RReverser** [comment]
> Thats awesome! do you want to send a followup applying all of those changes?

Hmm I rather posted that for inspiration, hoping to nerd-snipe you into trying the tool :) I'm sure it will come in handy in your efforts on modernising the codebase.

I guess I could send it later, but for now having enough on my plate as it is and time to sleep 😅

**Comment by sbc100** [comment]
> > Thats awesome! do you want to send a followup applying all of those changes?
> 
> Hmm I rather posted that for inspiration, hoping to nerd-snipe you into trying the tool :) I'm sure it will come in handy in your efforts on modernising the codebase.
> 
> I guess I could send it later, but for now having enough on my plate as it is and time to sleep 😅

Oh for sure i will try it out!  Thanks for suggesting..  I didn't know such a thing existed.   I'm just a humble grep user but this like of thing is way nicer.

**Comment by RReverser** [comment]
> I guess I could send it later, but for now having enough on my plate as it is and time to sleep 😅

Sent here. https://github.com/emscripten-core/emscripten/pull/20491

I just did minimal indentation fixes after rewrite, but otherwise don't want to spend too much time to format it. Feel free to edit the PR directly / merge / reject.

**Review by kripken — APPROVED** [review]

**Comment by sbc100** [comment]
Rebased on top of #20491

**Pull request merged**